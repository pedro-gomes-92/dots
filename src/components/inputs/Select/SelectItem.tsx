import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../../Base';

export interface SelectItemProps extends BaseProps {
  text: string;
  value?: FormValue;
}

export const SelectItem: FunctionComponent<SelectItemProps> = ({
  className,
  text,
  attribute,
  value,
  ...rest
}: SelectItemProps): JSX.Element => (
  <Base className={classnames('select-option', className)} attribute={{ ...attribute, value }} {...rest}>
    {text}
  </Base>
);

SelectItem.defaultProps = {
  tag: 'option',
};
