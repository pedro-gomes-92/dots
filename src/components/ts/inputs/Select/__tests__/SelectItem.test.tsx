import React from 'react';
import { render } from '@testing-library/react';

import { SelectItem } from '../SelectItem';

describe('Given <SelectItem />', (): void => {
  let container: HTMLElement;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<SelectItem text="Testing" />);
      expect(container).toMatchSnapshot();
    });

    describe('And uses value', (): void => {
      beforeEach((): void => {
        container = render(<SelectItem text="Testing" value="ts" />).container;
      });

      it('Then should match the snapshot', (): void => {
        expect(container).toMatchSnapshot();
      });

      it('And should update model', (): void => {
        expect((container.querySelector('option') as HTMLOptionElement).value).toEqual('ts');
      });
    });
  });
});
