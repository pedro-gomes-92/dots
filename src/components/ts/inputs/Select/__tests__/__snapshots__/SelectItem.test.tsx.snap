// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <SelectItem /> When initializes And uses value Then should match the snapshot 1`] = `
<div>
  <option
    class="select-option"
    value="ts"
  >
    Testing
  </option>
</div>
`;

exports[`Given <SelectItem /> When initializes Then should match the snapshot 1`] = `
<div>
  <option
    class="select-option"
  >
    Testing
  </option>
</div>
`;
